app/services/idv/data_url_image.rb

```diff
@@ -1,32 +1,28 @@
 module Idv
   class DataUrlImage
-    def initialize(data_url)
-      header, data = URI(data_url.chomp).opaque.to_s.split(',', 2)
-      @header = header.to_s
-      @data = data.to_s
-    end
-
-    BASE64_CONTENT_TYPE = /;base64$/.freeze
+    class InvalidUrlFormatError < StandardError; end
 
-    # @return [String]
-    def content_type
-      content_type, *_rest = @header.split(';')
-      content_type.to_s
+    def initialize(data_url)
+      header_and_data = data_url.delete_prefix('data:')
+      raise InvalidUrlFormatError.new if header_and_data == data_url
```

Comment:
I thought I was following the expected behavior that was happening with a URI constructor throwing on an invalid URL, but I suppose URI is more a method call than an instance constructor?

Wonder if it would be more sensible if it was part of a parse procedure, e.g.

```rb
def initialize(header:, data:)
  @header = header
  @data = data
end

def parse(data_url)
  header_and_data = data_url.delete_prefix('data:')
  raise InvalidUrlFormatError.new if header_and_data == data_url
  header, data = header_and_data.split(',', 2)
  DataUrlImage.new(header: header, data: data)
end
```

Reply:
I like that! But also, in the spirit of keeping changes small, what we have on the PR already is fine. the `URI` constructor throws an error for invalid data, so it follow that our constructor could too